src/components/hint/index.tsx

```diff
@@ -165,24 +166,38 @@ class Hint extends Component<HintProps, HintState> {
 
   targetRef: ElementRef<typeof RNView> | null = null;
   hintRef: ElementRef<typeof RNView> | null = null;
+  animationDuration = 170;
 
   state = {
     targetLayoutInWindow: undefined,
-    targetLayout: this.props.targetFrame
+    targetLayout: this.props.targetFrame,
+    animationEnded: false
   };
 
   visibleAnimated = new Animated.Value(Number(!!this.props.visible));
 
   componentDidUpdate(prevProps: HintProps) {
     if (prevProps.visible !== this.props.visible) {
-      Animated.timing(this.visibleAnimated, {
-        toValue: Number(!!this.props.visible),
-        duration: 170,
-        useNativeDriver: true
-      }).start();
+      this.animateHint();
     }
   }
 
+  animateHint = () => {
+    Animated.timing(this.visibleAnimated, {
+      toValue: Number(!!this.props.visible),
+      duration: this.animationDuration,
+      useNativeDriver: true
+    }).start(this.toggleAnimationEndedToRemoveHint);
+  };
+
+  toggleAnimationEndedToRemoveHint = () => {
+    this.setState({animationEnded: true}, () => {
+      setTimeout(() => {
+        this.setState({animationEnded: false});
+      }, this.animationDuration);
+    });
+  };
```

Comment:
This forces us to trigger two setState calls every time we trigger the animation. twice on enter twice on exit. 
Maybe we can reduce that.. 
If you think about it, while the Hint is visible it doesn't matter really what's the value of `animationEnded` state. 

What if instead of keeping `animationEnded` you'll keep something else, like `actuallyVisible`, `hasUnmounted` (or anything with a better name (: )
Then you'll update it once on enter and once on exit...

Reply:
Right, fixed